src/selectors/per-thread/markers.js

```diff
@@ -94,19 +97,53 @@ export function getMarkerSelectorsPerThread(threadSelectors: *) {
       )
   );
 
-  const getCommittedRangeFilteredMarkers: Selector<Marker[]> = createSelector(
-    getReferenceMarkerTable,
+  const getMarkerGetter: Selector<(MarkerIndex) => Marker> = createSelector(
+    getFullMarkerList,
+    markerList => (markerIndex: MarkerIndex): Marker => {
+      const marker = markerList[markerIndex];
+      if (!marker) {
+        throw new Error(stripIndent`
+          Tried to get marker index ${markerIndex} but it's not in the full list.
+          This is a programming error.
+        `);
+      }
+      return marker;
+    }
+  );
+
+  const getReferenceMarkerIndexes: Selector<MarkerIndex[]> = createSelector(
+    getFullMarkerList,
+    markers => markers.map((_, i) => i)
+  );
+
+  const filterMarkerIndexesCreator = (filterFunc: Marker => boolean) => (
```

Comment:
I'm not sure what this function is supposed to be doing, so perhaps it needs some documentation?

Reply:
yep, I'll add some!

Reply:
```js
  /* This utility function makes it easy to write selectors that deal with list
   * of marker indexes.
   * It takes a filtering function as parameter. This filtering function takes a
   * marker as parameter and returns a boolean deciding whether this marker
   * should be kept.
   * This function returns a function that does the actual filtering.
   *
   * It is typically used this way:
   *  const filteredMarkerIndexes = createSelector(
   *    getMarkerGetter,
   *    getSourceMarkerIndexesSelector,
   *    filterMarkerIndexesCreator(
   *      marker => MarkerData.isNetworkMarker(marker)
   *    )
   *  );
   */
```